src/renderer/mod.rs

```diff
@@ -70,6 +74,7 @@ impl Default for RendererSettings {
             debug_renderer: false,
             profiler: false,
             underline_stroke_scale: 1.,
+            group_zindex_step: 3,
```

Comment:
I there a reason why we need this, or at least why we need to have it as 3 as default. I would prefer if we could limit the effects to the same z-index. 

This might mean that some plugins have to change slightly, but if it's possible it's still beter IMO, since any hardcoded numbers have the potential to cause problems.

Reply:
In fact, this value could just be set to 2, as the z-index difference between the border and content windows in nui.nvim is 1. This parameter should be customizable, considering there may be cases where users layer windows with more z-index levels, and we should support such cases.